app/src/main/java/com/czterysery/ledcontroller/presenter/MainPresenterImpl.kt

```diff
@@ -196,26 +198,58 @@ class MainPresenterImpl(
 
             else ->
                 socketManager.connect(
-                        btController.getDeviceAddress(deviceName) as String,
-                        btController.adapter as BluetoothAdapter
+                    btController.getDeviceAddress(deviceName) as String,
+                    btController.adapter as BluetoothAdapter
                 ).subscribeOn(Schedulers.io())
-                        .subscribe({
-                            sendConnectionMessage(connected = true)
-                        }, { error ->
-                            view?.showLoading(shouldShow = false)
-                            Log.e(TAG, "Couldn't connect to device: $error")
-                        })
+                    .subscribe({
+                        sendConnectionMessage(connected = true)
+                        tryToGetConfiguration()
+                    }, { error ->
+                        view?.showLoading(shouldShow = false)
+                        Log.e(TAG, "Couldn't connect to device: $error")
+                    })
         }
     }
 
     private fun tryToGetConfiguration() {
-        // TODO `showLoading`
-        // TODO Here create get configuartion with 5sec or less timeout
-        // TODO `hideLoading`
+        view?.showLoading()
+        configurationListenerDisposable?.dispose()
+        configurationListenerDisposable =
+            Completable.fromCallable {
+                writeMessage(Messages.GET_CONFIGURATION)
```

Comment:
Unreadable flow. Write it like sendMessage andThen wait for response 2 or less sec